pkg/agents/team6/treaties.go

```diff
@@ -0,0 +1,375 @@
+package team6
+
+import (
+	"math"
+	"math/rand"
+
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/SOMAS2021/SOMAS2021/pkg/messages"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/food"
+)
+
+// We only accept LeaveAmountFood and LeavePercentFood treaties that say leave ≥,> of food available.
+// Hence there will never be a invalid treaty.
+
+// The utility function with
+// x - food input
+// z - desired food (maximum of the function)
+func calculateUtility(x, z float64, socialMotive string) float64 {
+
+	params := newUtilityParams(socialMotive)
+	// Calculates the function scaling parameter a
+
+	var result float64
+
+	if socialMotive == "Altruist" || socialMotive == "Nacissist" {
+		// Don't scale depending on desired food
+		result = params.g*math.Pow(x, 1/params.r) - params.c*x
+	} else {
+		a := (1 / z) * math.Pow((params.c*params.r)/params.g, params.r/(1-params.r))
+		result = params.g*math.Pow(a*x, 1/params.r) - params.c*a*x
```

Comment:
I see you set `params.r` to never be 0 in the other function, but just confirming that `r` will never be set to 0 or 1 during sim?

Reply:
Again, fair spot - these params are static (and we have a special case for when we have `params.r == 1`)

Reply:
What I'm seeing is that if `!(socialMotive == "Altruist" || socialMotive == "Nacissist)"`, regardless of the value of `r`, this is reached. If `r` is set to 1 what is the behavior? (because you get `params.r/(1-params.r) = 1/0`)

In all fairness, the only time the calculation would be valid is when `(params.c*params.r)/params.g` is 1, you then get `x^(x/(1-x))`

If `r` never reaches an illegal value, feel free to resolve the convo!

Reply:
this is because the value of r is specific to the behaviour (initialised in config!) - testing for behaviour also tests for r :)

Reply:
(reopening so you can see response)

Reply:
`r=1` is only the case for `socialMotive="Altruist"`. In this case, we go in the first `if` statement where we do not divide by `(1-r)`